Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/Batching/Batching.xaml.cs

```diff
@@ -0,0 +1,240 @@
+    {
+        private LearningModelSession nonBatchingSession_;
+        private LearningModelSession BatchingSession_;
+        float avgNonBatchedDuration = 0;
+        float avgBatchDuration = 0;
+        static bool navigatingAwayFromPage = false;
+
+        public Batching()
+        {
+            this.InitializeComponent();
+            // Ensure static variable is always false on page initialization
+            navigatingAwayFromPage = false;
+        }
+
+        async private void StartInference(object sender, RoutedEventArgs e)
+        {
+            ShowEvalUI();
+            ResetEvalMetrics();
+
+            var inputImages = GetInputImages();
+            int batchSize = GetBatchSizeFromBatchSizeComboBox();
+            CreateSessions(batchSize);
+
+            UpdateEvalText(false);
+            await Classify(inputImages);
+
+            UpdateEvalText(true);
+            await ClassifyBatched(inputImages, batchSize);
+
+            GenerateEvalResultAndUI();
+        }
+
+        private void ShowEvalUI()
+        {
+            StartInferenceBtn.IsEnabled = false;
+            EvalResults.Visibility = Visibility.Collapsed;
+            LoadingContainer.Visibility = Visibility.Visible;
+        }
+
+        private void ResetEvalMetrics()
+        {
+            avgNonBatchedDuration = 0;
+            avgBatchDuration = 0;
+        }
+
+        private List<VideoFrame> GetInputImages()
+        {
+            var birdFile = StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///InputData/hummingbird.jpg")).GetAwaiter().GetResult();
+            var catFile = StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///InputData/kitten.png")).GetAwaiter().GetResult();
+            var birdImage = CreateSoftwareBitmapFromStorageFile(birdFile);
+            var catImage = CreateSoftwareBitmapFromStorageFile(catFile);
+            var inputImages = new List<VideoFrame>();
+            for (int i = 0; i < 25; i++)
+            {
+                inputImages.Add(VideoFrame.CreateWithSoftwareBitmap(birdImage));
+                inputImages.Add(VideoFrame.CreateWithSoftwareBitmap(catImage));
+            }
+            return inputImages;
+        }
+
+        private SoftwareBitmap CreateSoftwareBitmapFromStorageFile(StorageFile file)
+        {
+            var stream = file.OpenAsync(FileAccessMode.Read).GetAwaiter().GetResult();
+            var decoder = BitmapDecoder.CreateAsync(stream).GetAwaiter().GetResult();
+            return decoder.GetSoftwareBitmapAsync().GetAwaiter().GetResult();
+        }
+
+        private void UpdateEvalText(bool isBatchingEval)
+        {
+            if (isBatchingEval)
+                EvalText.Text = "Inferencing Batched Inputs:";
+            else
+                EvalText.Text = "Inferencing Non-Batched Inputs:";
+        }
+
+        private void CreateSessions(int batchSizeOverride)
+        {
+            var modelPath = "ms-appx:///Models/squeezenet1.1-7.onnx";
```

Comment:
Is this sample targeting this model only? can we not hard-code the path here and switch to other models?

Reply:
Yes, this sample is just using the SqueezeNet model. We discussed possibly allowing for the selection of other models but we decided that it's not the core focus of the sample which is to showcase batching